polkadot/node/subsystem-types/src/messages.rs

```diff
@@ -204,6 +205,45 @@ pub enum CandidateValidationMessage {
 	},
 }
 
+/// Extends primitives::PvfExecKind, which is a runtime parameter we don't want to change,
+/// to separate and prioritize execution jobs by request type.
+/// The order is important, because we iterate through the values and assume it is going from higher
+/// to lowest priority.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, EnumIter)]
```

Comment:
Given that we are already leaking to validation what we want to have validated here, we should likely keep the old name `PvfExecKind` and infer "priority" from this, together with other information. E.g. whether a validation is still worthwhile (ttl PR based on this one).